packages/cli/src/ui/utils/textOutput.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// Internal state for the singleton
+let lastChar = '\n';
```

Comment:
nit: change this to 
bool 
atStartOfLine or column. Having this be '\n' to start is otherwise misleading.